spec-tests/src/test/java/org/candlepin/spec/pools/PoolUnlimitedMasterSpecTest.java

```diff
@@ -0,0 +1,373 @@
+import org.candlepin.dto.api.client.v1.EntitlementDTO;
+import org.candlepin.dto.api.client.v1.GuestIdDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.invoker.client.ApiException;;
+import org.candlepin.resource.client.v1.OwnerProductApi;
+import org.candlepin.spec.bootstrap.assertions.OnlyInStandalone;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.api.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+import org.candlepin.spec.bootstrap.data.builder.ConsumerTypes;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+@SpecTest
+public class PoolUnlimitedMasterSpecTest {
+
+    private static ApiClient client;
+    private OwnerDTO owner;
+    private static OwnerClient ownerApi;
+    private static OwnerProductApi ownerProductApi;
+    private static ConsumerClient consumerApi;
+    private ConsumerDTO physicalSystem;
+    private ApiClient physicalClient;
+    private ConsumerDTO guest;
+    private ApiClient guestClient;
+    private ConsumerDTO guestUnmapped;
+    private ApiClient guestUnmappedClient;
+    private ProductDTO productNoVirt;
+    private ProductDTO productUnlimitedVirt;
+    private ProductDTO productVirt;
+    private ProductDTO proudctVirtHostDep;
+    private ProductDTO productVirtInstanceMuliplier;
+    private ProductDTO productVirtProductMuliplier;
+    private PoolDTO poolNoVirt;
+    private PoolDTO poolUnlimitedVirt;
+    private PoolDTO poolVirt;
+    private PoolDTO poolVirtHostDep;
+    private PoolDTO poolVirtInstanceMuliplier;
+    private PoolDTO poolVirtProductMuliplier;
+
+    @BeforeAll
+    public static void beforeAll() throws ApiException {
+        client = ApiClients.admin();
+        ownerApi = client.owners();
+        ownerProductApi = client.ownerProducts();
+        consumerApi = client.consumers();
+    }
+
+    @BeforeEach
+    public void beforeEach() throws ApiException, InterruptedException {
+        owner = ownerApi.createOwner(Owners.random());
+        String guestUuid = StringUtil.random("uuid");
+
+        physicalSystem = Consumers.random(owner)
+            .name(StringUtil.random("host"))
+            .type(ConsumerTypes.System.value())
+            .guestIds(List.of(new GuestIdDTO().guestId(guestUuid)));
```

Comment:
Can we also remove the owner & consumers creation out of the before (maybe with a similar utility method as the one for products/pools? as per the guidelines we should only be creating data here if they are to be read-only, but the tests here are attaching entitlements to them